src/python/pants/engine/scheduler.py

```diff
@@ -41,70 +41,39 @@ class ExecutionRequest(datatype('ExecutionRequest', ['roots'])):
   """
 
 
-class LocalScheduler(object):
-  """A scheduler that expands a product Graph by executing user defined tasks."""
-
-  def __init__(self,
-               goals,
-               tasks,
-               project_tree,
-               native,
-               graph_lock=None):
-    """
-    :param goals: A dict from a goal name to a product type. A goal is just an alias for a
-           particular (possibly synthetic) product.
-    :param tasks: A set of (output, input selection clause, task function) triples which
-           is used to compute values in the product graph.
-    :param project_tree: An instance of ProjectTree for the current build root.
-    :param native: An instance of engine.subsystem.native.Native.
-    :param graph_lock: A re-entrant lock to use for guarding access to the internal product Graph
-                       instance. Defaults to creating a new threading.RLock().
-    """
-    self._products_by_goal = goals
-    self._project_tree = project_tree
-    self._native = native
-    self._product_graph_lock = graph_lock or threading.RLock()
-    self._run_count = 0
-
+class WrappedNativeScheduler(object):
```

Comment:
I'm definitely not a fan of this split. Seems like a huge amount of redundancy for minor benefit.